docs/inference-providers/guides/structured-output.md

```diff
@@ -0,0 +1,467 @@
+
+We'll create a simple schema that captures the most essential elements: the paper's title and a summary of its abstract. The easiest way to do this is to use Pydantic, a library that allows you to define Python classes that represent JSON schemas (among other things).
+
+<hfoptions id="json-pydantic">
+
+<hfoption id="pydantic">
+
+```python
+from pydantic import BaseModel
+
+class PaperAnalysis(BaseModel):
+    title: str
+    abstract_summary: str
+```
+
+</hfoption>
+
+<hfoption id="json">
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "title": {"type": "string"},
+    "abstract_summary": {"type": "string"}
+  },
+  "required": ["title", "abstract_summary"]
+}
+```
+
+</hfoption>
+
+</hfoptions>
+
+If you switch to the JSON option, you can see that the schema is exactly the same as the Pydantic model. This is what the LLM will receive as a response format instruction.
+
+This simple schema ensures we'll always get the paper's title and a concise summary of its abstract. Notice how we mark both fields as required - this guarantees they'll always be present in the response, making our application more reliable.
+
+## Step 2: Set up your inference client
+
+Now that we have our schema defined, let's set up the client to communicate with the inference providers. We'll show you two approaches: the Hugging Face Hub client (which gives you direct access to all Inference Providers) and the OpenAI client (which works through OpenAI-compatible endpoints).
+
+<hfoptions id="structured-outputs-implementation">
+
+<hfoption id="huggingface_hub">
+
+Install the Hugging Face Hub python package:
+
+```bash
+pip install huggingface_hub
+```
+
+Here's how to initialize the Hugging Face Hub client with the `InferenceClient` class.
+
+```python
+import os
+from huggingface_hub import InferenceClient
+
+# Initialize the client
+client = InferenceClient(
+    provider="cerebras",  # or use "auto" for automatic selection
+    api_key=os.environ["HF_TOKEN"],
+)
+
+```
+
+</hfoption>
+
+<hfoption id="openai">
+
+Install the OpenAI python package:
+
+```bash
+pip install openai
+```
+
+Here's how to initialize the OpenAI client with the `OpenAI` class.
+
+```python
+import os
+from openai import OpenAI
+from pydantic import BaseModel
+from typing import List
+
+# Initialize the OpenAI client (works with Inference Providers)
+client = OpenAI(
+    base_url="https://router.huggingface.co/cerebras/v1",
+    api_key=os.environ["HF_TOKEN"]
+)
+
+```
+
+</hfoption>
+
+</hfoptions>
+
+<Tip>
+
+Structured outputs are good use case for selecting a specific provider and model because you want to avoid incompatibility issues between the model, provider and the schema.
```

Comment:
```suggestion
Structured outputs are a good use case for selecting a specific provider and model because you want to avoid incompatibility issues between the model, provider and the schema.
```